### PR TITLE
LifecycleDeleteObjectTask: BackbeatRoutes in S3C do not support non versioned buckets

### PR DESCRIPTION
**Pull request by benzekrimaha**
Issue : BB-555

### CODE REVIEW

File: extensions/lifecycle/tasks/LifecycleDeleteObjectTask.js [hunk 1]
```diff
@@ -47,7 +47,12 @@ class LifecycleDeleteObjectTask extends BackbeatTask {
         }, log, (err, blob) => {
             LifecycleMetrics.onS3Request(log, 'getMetadata', 'expiration', err);
             if (err) {
-                log.error('error getting metadata blob from S3', Object.assign({
+                // <!> Only in S3C <!> Backbeat API returns 'InvalidBucketState' error if the bucket is not versioned.
+                // In this case, instead of logging an error, it should be logged as a debug message,
+                // to avoid causing unnecessary concern to the customer.
+                // TODO: After the implementation of CLDSRV-461, we could remove this check.
```
Comment (ghost) [inline extensions/lifecycle/tasks/LifecycleDeleteObjectTask.js]: Let's have a ticket number with the TODO :slightly_smiling_face:

File: tests/unit/mocks.js [hunk 2]
```diff
@@ -79,6 +79,7 @@ class BackbeatClientMock {
         this.response = { error, data };
     }
 
+
```
Comment (ghost) [inline tests/unit/mocks.js]:
```suggestion
```

**Review by welansari — APPROVED** [review]

File: extensions/lifecycle/tasks/LifecycleDeleteObjectTask.js (hunk 1 above)
Comment (benzekrimaha) [reply extensions/lifecycle/tasks/LifecycleDeleteObjectTask.js]: Done here : https://github.com/scality/backbeat/pull/2541/files#diff-0a199af7ee10e4c59a4d84364fad46f84ce9c1cd9bffe7b55c54ed03eddb5166R53

**Review by ghost — APPROVED** [review]

**Pull request merged**